From 07108903c5320b83188837f8233d19949ca28319 Mon Sep 17 00:00:00 2001
From: Max Rottenkolber <maximilian@igalia.com>
Date: Mon, 10 Mar 2025 03:47:52 -0700
Subject: [PATCH] [JSC] Fix integer overflow on armv7 in
 CompleteSubspace::tryAllocateSlow
 https://bugs.webkit.org/show_bug.cgi?id=289157

Reviewed by Yusuke Suzuki.

Avoids integer overflow when multiplying by ramSize.

* Source/JavaScriptCore/heap/CompleteSubspace.cpp:
(JSC::CompleteSubspace::tryAllocateSlow):

Canonical link: https://commits.webkit.org/291887@main
---
 Source/JavaScriptCore/heap/CompleteSubspace.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/JavaScriptCore/heap/CompleteSubspace.cpp b/Source/JavaScriptCore/heap/CompleteSubspace.cpp
index 0c41b4ea0b47..62c11706e3d0 100644
--- a/Source/JavaScriptCore/heap/CompleteSubspace.cpp
+++ b/Source/JavaScriptCore/heap/CompleteSubspace.cpp
@@ -130,7 +130,7 @@ void* CompleteSubspace::tryAllocateSlow(VM& vm, size_t size, GCDeferralContext*
     
     vm.heap.collectIfNecessaryOrDefer(deferralContext);
     if (UNLIKELY(Options::maxHeapSizeAsRAMSizeMultiple())) {
-        if (vm.heap.capacity() > Options::maxHeapSizeAsRAMSizeMultiple() * WTF::ramSize())
+        if (vm.heap.capacity() > static_cast<uint64_t>(Options::maxHeapSizeAsRAMSizeMultiple()) * static_cast<uint64_t>(WTF::ramSize()))
             return nullptr;
     }
 
-- 
2.34.1

